s4e-backend/src/main/java/pl/cyfronet/s4e/sync/PipelineFactory.java

```diff
@@ -0,0 +1,164 @@
+package pl.cyfronet.s4e.sync;
+
+import lombok.RequiredArgsConstructor;
+import org.leadpony.justify.api.JsonValidationService;
+import org.springframework.stereotype.Component;
+import pl.cyfronet.s4e.bean.Schema;
+import pl.cyfronet.s4e.data.repository.ProductRepository;
+import pl.cyfronet.s4e.data.repository.SchemaRepository;
+import pl.cyfronet.s4e.service.SceneStorage;
+import pl.cyfronet.s4e.sync.context.Context;
+import pl.cyfronet.s4e.sync.step.*;
+import pl.cyfronet.s4e.sync.step.metadata.*;
+import pl.cyfronet.s4e.util.GeometryUtil;
+
+import java.util.List;
+import java.util.Map;
+
+@Component
+@RequiredArgsConstructor
+public class PipelineFactory {
+    public static final String SCENE_METADATA_ARTIFACT = "metadata";
+
+    private final SceneStorage sceneStorage;
+    private final SchemaRepository schemaRepository;
+    private final ProductRepository productRepository;
+    private final GeometryUtil geometryUtil;
+    private final ScenePersister scenePersister;
+    private final JsonValidationService jsonValidationService = JsonValidationService.newInstance();
+
+    public List<Step<Context, Error>> build() {
+        return List.of(
+                ResetErrorParameters.builder()
+                        .parameters(c -> Map.of(
+                                "file_type", "scene",
+                                "file_key", c.getScene().getKey()))
+                        .build(),
+                // Check scene file exists.
+                LoadKeyContent.builder()
+                        .sceneStorage(() -> sceneStorage)
+                        .key(c -> c.getScene().getKey())
+                        .update((c, content) -> c.getScene().setContent(content))
+                        .build(),
+                // Check scene schema exists and is of correct type.
```

Comment:
Is those blocks of code going to be splited to methods?

Reply:
The actual implementation is already split into steps, and they form a single pipeline. Do you think it could be improved somehow?

Reply:
I think that instead of comments you can use methods that are running those blocks of code